### PR TITLE
Check the confidence levels of a transcript  with minConfidence

### PR DESCRIPTION
**Pull request by rammohan-kore**
https://github.com/jambonz/jambonz-feature-server/issues/807
Check whether the confidence levels of a transcript is greater than or equal to the minConfidence property set in the gather verb, if confidence level of the transcript is less then resolve it as a timeout and proceed with next verb

### CODE REVIEW

**Review by davehorton — CHANGES_REQUESTED** [review]
please refer to latest speech-utils, where minConfidence is a property on recognizer.  Also this feature should work both for gather and config (ie to set a session-level default).

File: lib/tasks/gather.js [hunk 1]
```diff
@@ -736,6 +736,18 @@ class TaskGather extends SttTask {
...
+    const minConfidence = this.data?.recognizer?.minConfidence;
+    this.logger.debug({evt},
+      `TaskGather:_onTranscription - confidence (${confidence}), minConfidence (${minConfidence})`);
+    if (confidence && minConfidence && confidence < minConfidence) {
+      this.logger.info({evt},
+        'TaskGather:_onTranscription - rejecting transcript as confidence ' +
+        `(${confidence}) < minConfidence (${minConfidence})`);
+      evt.error = 'Rejecting transcript as confidence ' +
+        `(${confidence}) < minConfidence (${minConfidence})`
+      return this._resolve('stt-min-confidence-error', evt);
```
Comment (davehorton) [inline lib/tasks/gather.js:749]: could we rename this event as `stt-low-confidence`?  Also, I don't really consider this an "error", as nothing necessarily went wrong.

File: lib/tasks/gather.js [hunk 2]
```diff
@@ -1091,6 +1103,12 @@ class TaskGather extends SttTask {
           this.emit('stt-error', evt);
           returnedVerbs = await this.performAction({reason: 'error', details: evt.error});
         }
+      } else if (reason.startsWith('stt-min-confidence-error')) {
+        if (this.parentTask) this.parentTask.emit('stt-min-confidence-error', evt);
+        else {
+          this.emit('stt-min-confidence-error', evt);
+          returnedVerbs = await this.performAction({reason: 'error', details: evt.error});
```
Comment (davehorton) [inline lib/tasks/gather.js:1110]: I do not think we should be sending a webhook with reason 'error'.  This is not an error -- errors are when things fail, this is simply a case of not getting very recognizable audio.

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/tasks/gather.js (hunk 1 above)
Comment (rammohan-kore) [reply lib/tasks/gather.js]: Yes, renamed stt-min-confidence-error as stt-min-confidence

File: lib/tasks/gather.js [hunk 3]
```diff
@@ -736,6 +736,16 @@ class TaskGather extends SttTask {
       this.logger.info({evt}, 'TaskGather:_onTranscription - got empty transcript, continue listening');
       return;
     }
+    const confidence = evt.alternatives[0].confidence;
+    const minConfidence = this.data?.recognizer?.minConfidence;
```
Comment (davehorton) [inline lib/tasks/gather.js:740]: minor point, but please remove ? after this.data - there is always a 'this.data' object

Comment (rammohan-kore) [reply lib/tasks/gather.js]: Remove ? after this.data

**Comment by davehorton** [comment]
please rebase to latest main

**Comment by davehorton** [comment]
also please run `npm run jslint` to make sure there are no linting errors

**Comment by rammohan-kore** [comment]
Rebased with main branch and ran jslint

**Pull request merged**